### PR TITLE
Add acme clients internal data structures and adjust tests

### PR DESCRIPTION
**Pull request by raskchanky**
https://hashicorp.atlassian.net/browse/VAULT-24625

I applied the diff on the linked Jira and adjusted a few tests. Some of the tests mentioned in the Jira didn't look like they needed adjusting? Or maybe I misread them?

### CODE REVIEW

**Comment by raskchanky** [comment]
I'm thinking maybe no changelog for this PR, since it's just internal changes?

File: vault/activity_log_test.go [hunk 1]
```diff
@@ -4267,43 +4267,44 @@ func TestActivityLog_processNewClients_delete(t *testing.T) {
...
+		require.Contains(t, byNS, ns)
+		require.Contains(t, byNS[ns].Mounts, mount)
 		monthIndex := timeutil.StartOfMonth(startTime).UTC().Unix()
 		require.Contains(t, byMonth, monthIndex)
 		require.Equal(t, byMonth[monthIndex].Namespaces, byNS)
 		require.Equal(t, byMonth[monthIndex].NewClients.Namespaces, byNS)
 
-		for _, typ := range []string{nonEntityTokenActivityType, secretSyncActivityType, entityActivityType} {
-			if clientType == typ || (clientType == ACMEActivityType && typ == nonEntityTokenActivityType) {
+		for _, typ := range []string{nonEntityTokenActivityType, secretSyncActivityType, entityActivityType, ACMEActivityType} {
```
Comment (miagilepner) [inline vault/activity_log_test.go:4293]: nit: this can be replaced with `ActivityClientTypes`

Comment (raskchanky) [reply vault/activity_log_test.go]: Fixed in https://github.com/hashicorp/vault/pull/26020/commits/cc3a80f4dc715611612b983d6b448e978335eb5e

**Comment by raskchanky** [comment]
@miagilepner Thanks for catching my changelog oversight! Added in https://github.com/hashicorp/vault/pull/26020/commits/d9987d6dbe913e6b13f0bdf6af5845d08c0f13e0

**Review by miagilepner — APPROVED** [review]
LGTM!

**Pull request merged**